File: leetcode/sliding-window/cpp/1493.longest-subarray-of-1-s-after-deleting-one-element.cpp
/*
 * @lc app=leetcode id=1493 lang=cpp
 *
 * [1493] Longest Subarray of 1's After Deleting One Element
 */

// @lc code=start
class Solution {
public:
	int longestSubarray(vector<int>& nums) {
		int	zeroIdx = -1, len = nums.size(), i = 0, maxSubarray = -1;

		for (int j = 0; j < len; j++) {
			if (nums[j] == 0) {
				i = zeroIdx + 1;
				zeroIdx = j;
			}
			maxSubarray = max(maxSubarray, j - i);
		}
		return maxSubarray;
	}
};
// @lc code=end

File: leetcode/sliding-window/py/1493.longest-subarray-of-1-s-after-deleting-one-element.py
#
# @lc app=leetcode id=1493 lang=python3
#
# [1493] Longest Subarray of 1's After Deleting One Element
#

# @lc code=start
class Solution:
	def longestSubarray(self, nums: List[int]) -> int:
		zeroIdx, size, i, maxSubarray = -1, len(nums), 0, -1
		for j in range(size):
			if nums[j] == 0:
				i = zeroIdx + 1
				zeroIdx = j
			maxSubarray = max(maxSubarray, j - i)
		return maxSubarray
# @lc code=end

